made edits (#269)
mulesoft/docs-api-manager: modules/ROOT/pages/header-removal-policy.adoc

= Header Removal
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

[width="100%", cols="5,15"]
|===
>s| Policy Name | Header Removal
>s|Summary      | Removes headers from a request or a response
>s|Category | Transformation
>s| First Mule version available | 4.1.0
.1+>.^s| Returned Status Codes
|
|===

> //any ret codes to add here Mariano?

Reviewer: @JaatadiaMulesoft what happens if an expression fails?

The Header Removal policy removes all listed headers from a request or a response of a message. When you configure this policy for your API, you must specify an inbound and outbound map of the headers that you want to remove from the message processing in the form of a key-value pair.

image::header-inject-remove-diagram.png[]

//Mariano can we add some text about what's happening in this diagram?

> You can optionally include Mule runtime engine (Mule) expressions for Mule 3.0 and DataWeave expressions for Mule 4.0 in the value of the header:

Reviewer: Same as before, we should prioritize mule 4 (it should be listed first)

Author: Done! Thanks!

`“#[attributes.requestPath]”`

> //We need some more details 

Reviewer: All headers matching the configured expression will be removed from the message.

Author: Done! Thanks!

== How This Policy Works

> The `content-type` and `content-length` headers of a request can be filtered by this policy, as well as the host header. In the response, only the `content-type` header is removed.

Reviewer: We should add that this is an example.

Author: Done! Thanks!

// image::remove-inbound-header.png[]
// +
// image::remove-outbound-header.png[]

> //Anything to add about what to keep in mind or check for before removing a header?    

Reviewer: Be careful when defining an expression since every header matching the expression will be removed.

Author: Done! Thanks!

== Configuring Policy Parameters

When you apply the Header Removal policy to your API from the UI, you can configure the following parameters:

[%header%autowidth.spread,cols="a,a"]
|===
| Parameter | Description 
|Inbound Header Map  | The key-value pair for the inbound header, for example, `in-header(key):3(value)`
|Outbound Header Map | The key-value pair for the outbound header, for example, `out-header(key):4(value)`
|Method & Resource conditions |The option to add configurations to only a select few or all methods and resources of the API

> |===

Reviewer: This example is incorrect for the removal policy, values are not provided, only the header name or expression

> = Header Removal

Reviewer 2: All edits made to the previous topic are applicable to this one as well. Please apply them throughout.
Copyedited only.

Reviewer 2: ```suggestion
= Header Removal Policy
```

Author: Done! Thanks!

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

[width="100%", cols="5,15"]
|===

> >s| Policy Name | Header Removal

Reviewer 2: ```suggestion
>s| Policy name | Header Removal
```

Author: Done! Thanks!

>s|Summary      | Removes headers from a request or a response 
>s|Category | Transformation
>s| First Mule version available | 4.1.0
.1+>.^s| Returned Status Codes

[%autowidth.spread,cols="a,a"]
!===
.1+>.^! 500 ! Configuration includes an expression that was not evaluated.
! The provided expression in the configuration was not validated. +
!===

|===

> The Header Removal policy removes all listed headers from a request or a response of a message. When you configure this policy for your API, you must specify an inbound and outbound map of the headers that you want to remove from the message processing in the form of a key-value pair.

Reviewer 2: ```suggestion
The Header Removal policy removes message request or response headers as indicated by the key-value maps specified:
```

Author: Done! Thanks!

image::header-inject-remove-diagram.png[]

You can optionally include DataWeave expressions for Mule 4.0 and Mule runtime engine (Mule) expressions for Mule 3.0 and in the value of the header:

`“#[attributes.requestPath]”`

In this example, all the headers matching the configured expression are removed from the message.

== How This Policy Works

> Every header matching an expression that you specify in the policy parameter is removed from a message. Therefore, you must double-check your expression for the header removal.
> 
> For example, the `content-type` and `content-length` headers and the host header of a request can be filtered by this policy. In the response, only the `content-type` header is removed:

Reviewer 2: ```suggestion
Because every header that matches a Header Removal policy parameter is removed from the message, you must be careful about how you configure that parameter. 

For example, the following parameter configuration removes only the `content-type` header from a message that also contains `content-length` and host headers:
```

Author: Done! Thanks!

.Inbound Header
image::remove-inbound-header.png[]
+
.Outbound Header
image::remove-outbound-header.png[]

== Configuring Policy Parameters

When you apply the Header Removal policy to your API from the UI, you can configure the following parameters:

[%header%autowidth.spread,cols="a,a"]
|===
| Parameter | Description 
|Inbound Header Map  | The key-value pair for the inbound header, for example, `in-header(key):3(value)`
|Outbound Header Map | The key-value pair for the outbound header, for example, `out-header(key):4(value)`
|Method & Resource conditions |The option to add configurations to only a select few or all methods and resources of the API

> |===

Reviewer 2: ditto

Author: Done! Thanks!
